aptos-move/framework/aptos-token-objects/sources/token.move

```diff
@@ -229,7 +287,19 @@ module aptos_token_objects::token {
         borrow(&token).description
     }
 
+    /// This method allows minting to happen in parallel, making it efficient.
+    fun name_snapshot<T: key>(token: &Object<T>): AggregatorSnapshot<String> acquires Token, TokenConcurrentFieldsAppendix {
+        let token_address = object::object_address(token);
+        if (exists<TokenConcurrentFieldsAppendix>(token_address)) {
+            aggregator_v2::copy_snapshot(&borrow_global<TokenConcurrentFieldsAppendix>(token_address).name)
+        } else {
+            aggregator_v2::create_snapshot(borrow(token).name)
+        }
+    }
+
     #[view]
+    /// Avoid this method in the same transaction as the token is minted
+    /// as that would prohibit transactions to be executed in parallel.
     public fun name<T: key>(token: Object<T>): String acquires Token {
         borrow(&token).name
     }
```

Comment:
since name is a deprecated field, we should probably check to see first if the new field exists and return that and then return name otherwise.

Reply:
Omg :/ nice catch. 

By the way, this is the PR that we might potentially deploy as a point release.

Full PR that we want to go out with 1.8 is this one, if you have time to take a look - https://github.com/aptos-labs/aptos-core/pull/9971/files (it has the same bug, of course :/ )